include/ur_client_library/comm/tcp_socket.h

```diff
@@ -59,7 +59,7 @@ class TCPSocket
   }
   virtual void setOptions(int socket_fd);
 
-  bool setup(std::string& host, int port);
+  bool setup(std::string& host, int port, size_t max_num_tries = 0);
```

Comment:
I know it is not related to this PR, but the host is parsed as a non const reference even though it is not modified within the function. I think it would make sense to add the const keyword to host and all the other parameters as they are not modified within the function. 
```suggestion
  bool setup(const std::string& host, const int port, const size_t max_num_tries = 0);
```